src/Maestro/tests/Maestro.ScenarioTests/MaestroScenarioTestBase.cs

```diff
@@ -100,25 +104,98 @@ public async Task<bool> WaitForMergedPullRequestAsync(string targetRepo, string
+        private async Task<int> GetAzDoPullRequestIdAsync(string targetRepoName, string targetBranch)
         {
+            string searchBaseUrl = GetAzDoRepoUrl(targetRepoName);
+            string apiBaseUrl = GetAzDoApiRepoUrl(targetRepoName);
+            IEnumerable<int> prs = new List<int>();
+
             int attempts = 10;
             while (attempts-- > 0)
             {
-                var prs = await AzDoClient.SearchPullRequestsAsync(targetRepoUri, targetBranch, Microsoft.DotNet.DarcLib.PrStatus.Open).ConfigureAwait(false);
+                try
+                {
+                    prs = await SearchPullRequestsAsync(searchBaseUrl, targetBranch).ConfigureAwait(false);
+                }
+                catch (HttpRequestException ex)
+                {
+                    if (ex.Message == "Response status code does not indicate success: 404 (Not Found).")
+                    {
+                        // This is expected before the PR has been created
+                    }
+                    else
+                    {
+                        throw;
+                    }
+                }
+
                 if (prs.Count() == 1)
                 {
-                    return await AzDoClient.GetPullRequestAsync($"{targetRepoUri}/pullrequests/{prs.FirstOrDefault()}?api-version=5.0");
+                    return prs.FirstOrDefault();
                 }
+
                 if (prs.Count() > 1)
                 {
-                    throw new MaestroTestException($"More than one pull request found in {targetRepoUri} targeting {targetBranch}");
+                    throw new MaestroTestException($"More than one pull request found in {targetRepoName} targeting {targetBranch}");
+                }
+
+                await Task.Delay(60 * 1000).ConfigureAwait(false);
+            }
+
+            throw new MaestroTestException($"No pull request was created in {searchBaseUrl} targeting {targetBranch}");
+        }
+
+        private async Task<IEnumerable<int>> SearchPullRequestsAsync(string repoUri, string targetPullRequestBranch)
+        {
+            (string accountName, string projectName, string repoName) = Microsoft.DotNet.DarcLib.AzureDevOpsClient.ParseRepoUri(repoUri);
+            var query = new StringBuilder();
+
+            Microsoft.DotNet.DarcLib.AzureDevOpsPrStatus prStatus = Microsoft.DotNet.DarcLib.AzureDevOpsPrStatus.Active;
+            query.Append($"searchCriteria.status={prStatus.ToString().ToLower()}");
+            query.Append($"&searchCriteria.targetRefName=refs/heads/{targetPullRequestBranch}");
+
+            JObject content = await _parameters.AzDoClient.ExecuteAzureDevOpsAPIRequestAsync(
+                HttpMethod.Get,
+                accountName,
+                projectName,
+                $"_apis/git/repositories/{repoName}/pullrequests?{query}",
+                new NUnitLogger()
+                );
+
+            JArray values = JArray.Parse(content["value"].ToString());
```

Comment:
Why are we doing a ToString and then Parsing?  Why not use the value that we already have?

Reply:
This is parsing from a JObject to something that can be enumerated

Reply:
content.Value<JArray>("value") is more clear and will work the same.

Reply:
Yes, `context["value"]` already _is_ a JArray... this code is taking that JArray, serializing it to a string, and then parsing that string back into another copy of that JArray.

Reply:
Huh. I wonder where I got this code from then, usually when I have something like this that's silly it's either because I found it somewhere else or it didn't work in an obvious way when I initially wrote the code. I'll look around and make sure I didn't just grab it from somewhere else in the code that's also doing extra parsing.